cli/sign_test.go

```diff
@@ -324,3 +324,40 @@ func TestSignWithWriteCommand(t *testing.T) {
 		assert.Contains(t, artifactTarLines, "manifest.sig")
 	})
 }
+
+func TestSignExistingPermissions(t *testing.T) {
+	// first create archive, that we will be able to read
+	updateTestDir, _ := ioutil.TempDir("", "update")
+	defer os.RemoveAll(updateTestDir)
+
+	priv, pub, err := generateKeys()
+	assert.NoError(t, err)
+
+	err = WriteArtifact(updateTestDir, 2, "")
+
+	err = MakeFakeUpdateDir(updateTestDir,
+		[]TestDirEntry{
+			{
+				Path:    "private.key",
+				Content: priv,
+				IsDir:   false,
+			},
+			{
+				Path:    "public.key",
```

Comment:
Suggestion: Since the public key is not used anywhere in the test. It is just really noise. I would suggest removing it.

Another option would be to also validate the signed artifact. In which case the public key would be useful :)

I would vote for the second option, but I will let you decide.